### PR TITLE
feat: scheduler (5/): add parallelizer

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR is part of the PRs that implement the Fleet workload scheduling.

It features a parallelizer, which the scheduler frameworks uses to run tasks in parallel.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- [x] Unit tests

### CODE REVIEW

File: pkg/scheduler/framework/parallelizer/errorflag.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package parallelizer
+
+// ErrorFlag is a flag for indicating whether an error has occurred when running tasks in
+// parallel with the parallelizer.
+type ErrorFlag struct {
```
Comment (ryanzhang-oss) [inline pkg/scheduler/framework/parallelizer/errorflag.go:10]: how does this work with Parallelizer?

Comment (michaelawyu) [reply pkg/scheduler/framework/parallelizer/errorflag.go:10]: Hi Ryan! The error flag is designed to be used in doWork when calling parallelizer.ParallelizeUntil -> it allows exit on first error (in combination with context `cancel()`).

**Review by ryanzhang-oss — DISMISSED** [review]

File: pkg/scheduler/framework/parallelizer/errorflag.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package parallelizer
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/parallelizer/errorflag.go:6]: add a package comment?

**Review by zhiying-lin — APPROVED** [review]
one minor comment, others LGTM

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**